### PR TITLE
Event registration

### PR DESCRIPTION
**Pull request by mxsnyder**
Created event reg form and services.

### CODE REVIEW

File: dist/js/apps/events-page/app.js [hunk 1]
```diff
@@ -280,6 +315,58 @@ var _typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol
...
+		var vm = this;
+
+		vm.isGroup = 'false';
+		vm.isSubmitted = false;
+		vm.isLoadingResults = false;
+
+		vm.submitHandler = function () {
+			vm.isLoadingResults = true;
+
+			var postModel = {
```
Comment (martypowell) [inline dist/js/apps/events-page/app.js]: Are there going to be more keys on this object? If no,t you could probably remove the outer

File: js/apps/events-page/controllers/eventRegistrationCtrl.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		const vm = this;
+
+		vm.isGroup = 'false';
+		vm.isSubmitted = false;
+		vm.isLoadingResults = false;
+
+		vm.submitHandler = () => {
+			vm.isLoadingResults = true;
+
+			const postModel = {
```
Comment (martypowell) [inline js/apps/events-page/controllers/eventRegistrationCtrl.js]: if there isnt going to be any other postModels remove one of the outer object

File: js/apps/events-page/dataServices/registrationService.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+((app) => {
+	'use strict';
+
+	const registrationService = (CONSTANTS, $http, $q) => {
+		const register = (registrationModel) => {
+			return $q((resolve, reject) => {
```
Comment (martypowell) [inline js/apps/events-page/dataServices/registrationService.js]: Use $q like this, https://docs.angularjs.org/api/ng/service/$q#testing create a deferred variable and return the deferred promise.

File: dist/js/apps/events-page/app.js (hunk 1 above)
Comment (mxsnyder) [reply dist/js/apps/events-page/app.js]: That's how the vendor API wants it, wrapped in another object. However, you have a point. I just have to make some small changes to the web service to match.

File: js/apps/events-page/controllers/eventRegistrationCtrl.js (hunk 2 above)
Comment (mxsnyder) [reply js/apps/events-page/controllers/eventRegistrationCtrl.js]: That's how the vendor API wants it, wrapped in another object. However, you have a point. I just have to make some small changes to the web service to match.

Comment (martypowell) [reply js/apps/events-page/controllers/eventRegistrationCtrl.js]: 10-4

File: dist/js/apps/events-page/app.js (hunk 1 above)
Comment (martypowell) [reply dist/js/apps/events-page/app.js]: 10-4

File: js/apps/events-page/dataServices/registrationService.js (hunk 3 above)
Comment (mxsnyder) [reply js/apps/events-page/dataServices/registrationService.js]: I'm skipping this for now, but I'll think about it and possibly address later. I don't see extra value in using `$q.defer()` over `$q((resolve, reject) => { ... }`, but I'm happy to discuss.

Comment (martypowell) [reply js/apps/events-page/dataServices/registrationService.js]: @mxsnyder no problem, I think it's slightly cleaner (my preference).

**Pull request merged**